src/main/java/org/scijava/search/module/ModuleSearcher.java

```diff
@@ -80,22 +81,33 @@ public String title() {
 			.collect(Collectors.toList());
 
 		final String textLower = text.toLowerCase();
+		final List<String> textLowerParts = Arrays.asList(textLower.split("\\s+"));
 
-		// First, add modules where title starts with the text.
+		// Add modules where title starts with the text.
 		modules.stream() //
 			.filter(info -> startsWith(info, textLower) ) //
 			.forEach(matches::add);
 
-		// Next, add modules where title has text inside somewhere.
+		// Add modules where title has text inside somewhere.
 		modules.stream() //
-			.filter(info -> hasSubstring(info, textLower)) //
+			.filter(info -> hasSubstringInTitle(info, textLower)) //
 			.forEach(matches::add);
 
-		// Finally, add modules where menu path has text inside somewhere.
+		// Add modules where menu path has text inside somewhere.
 		modules.stream() //
-			.filter(info -> hasSubstring(info, textLower)) //
+			.filter(info -> hasSubstringInMenu(info, textLower)) //
 			.forEach(matches::add);
 
+		// Add modules where title has all parts of the text inside somewhere.
+		modules.stream() //
+				.filter(info -> hasSubstringsInTitle(info, textLowerParts)) //
+				.forEach(matches::add);
```

Comment:
I didn't test this, but do we actually make sure that no module is added _multiple_ times to _matches_? As the code is currently, there's a possibility that a module matches two or more of these filters, and therefore might get added twice, no? I'd like to avoid duplication in the search result to avoid confusion.
(Please ignore if my analysis is incorrect :)

Reply:
Good old `LinkedHashSet`. Such a convenient data structure! 😃